File: 2024/01/01.py
pairs = [list(map(int, line.strip().split())) for line in open("input.txt")]
left = [p[0] for p in pairs]
right = [p[1] for p in pairs]
left.sort()
right.sort()
out = 0
right_count = {}
for i in range(len(left)):
    out += abs(left[i]-right[i])
    if right[i] not in right_count:
        right_count[right[i]] = 0
    right_count[right[i]] += 1
out2 = 0
for i in left:
    out2 += i*right_count.get(i, 0)
print(out)
print(out2)

File: 2024/01/01.cpp
#include "../../util.hpp"

int main() {
    auto v = file_to_v();
    const int n = v.size();
    vector<int> left(n), right(n);
    for (int i=0; i<n; i++) {
        auto l = split(v[i]);
        left[i] = stoi(l[0]);
        right[i] = stoi(l[1]);
    }
    sort(left.begin(), left.end());
    sort(right.begin(), right.end());
    int out1 = 0, out2 = 0;
    map<int, int> c1, c2;
    for (int i=0; i<n; i++) {
        out1 += abs(left[i]-right[i]);
        ++c1[left[i]];
        ++c2[right[i]];
    }
    for (auto& [key, val] : c1) out2 += key*val*c2[key];
    cout << out1 << "\n" << out2 << endl;
    return 0;
}